### ISSUE TITLE
`ResourceGridDemapper` layer contains some bugs.

### ISSUE THREAD

**Issue by kassankar** [NVlabs/sionna#19]
**Describe the bug**
The layer `ResourceGridDemapper` does not work properly.

**To Reproduce**
Please, find below a minimal code to reproduce the error:
```
import tensorflow as tf
from tensorflow.keras import Model
from sionna.ofdm import ResourceGrid,ResourceGridDemapper
from sionna.mimo import StreamManagement
from sionna.utils import QAMSource
import numpy as np

class E2ESystem(Model):
    def __init__(self):
        super().__init__()
        self.cp_length = 0
        self.fft_size = 72
        self.num_ofdm_symbols = 14
        self.num_streams_per_tx = 1
        self.subcarrier_spacing = 15e3
        self.stream_manager = StreamManagement(np.array([[1]]),1)
        self.rg = ResourceGrid(num_ofdm_symbols = self.num_ofdm_symbols,
                  fft_size = self.fft_size,
                  subcarrier_spacing = self.subcarrier_spacing,
                  num_tx = 1,
                  num_streams_per_tx = self.num_streams_per_tx)
        self.qam = QAMSource(4)
        self.rg_demap  = ResourceGridDemapper(self.rg,self.stream_manager)

    
    @tf.function
    def call(self, batch_size):
        x_rg = self.qam([batch_size, 1, 1, self.num_ofdm_symbols, self.fft_size])
        print(x_rg.shape)
        x_rg_dem = self.rg_demap(x_rg)
        return x_rg_dem

e2e = E2ESystem()
e2e(128)

```

**Expected behavior**
The `ResourceGridDemapper` layer should return the data mapped from the `qam` layer.

**Proposed solution**
In the `ResourceGridDemapper` layer replace : 
    1. `if tf.rank(y)==5:` --> `if len(y.shape)==5:` 
    2. `if tf.shape(y)[-1]==1:` --> `if y.shape[-1]==1:`
Please, find below the new version of the layer:
```
class _ResourceGridDemapper(Layer):
    # pylint: disable=line-too-long
    r"""ResourceGridDemapper(resource_grid, stream_management, dtype=tf.complex64, **kwargs)

    Extracts data-carrying resource elements from a resource grid.

    This layer takes as input an OFDM :class:`~sionna.ofdm.ResourceGrid` and
    extracts the data-carrying resource elements. In other words, it implements
    the reverse operation of :class:`~sionna.ofdm.ResourceGridMapper`.

    Parameters
    ----------
    resource_grid : ResourceGrid
        An instance of :class:`~sionna.ofdm.ResourceGrid`.

    stream_management : StreamManagement
        An instance of :class:`~sionna.mimo.StreamManagement`.

    dtype : tf.Dtype
        Datatype for internal calculations and the output dtype.
        Defaults to `tf.complex64`.

    Input
    -----
    : [batch_size, num_rx, num_streams_per_rx, num_ofdm_symbols, fft_size, data_dim]
        The full OFDM resource grid in the frequency domain.
        The last dimension `data_dim` is optional. If `data_dim`
        is used, it refers to the dimensionality of the data that should be
        demapped to individual streams. An example would be LLRs.

    Output
    ------
    : [batch_size, num_rx, num_streams_per_rx, num_data_symbols, data_dim]
        The data that were mapped into the resource grid.
        The last dimension `data_dim` is only returned if it was used for the
        input.
    """
    def __init__(self,
                 resource_grid,
                 stream_management,
                 dtype=tf.complex64,
                 **kwargs):
        super().__init__(dtype=dtype, **kwargs)
        self._stream_management = stream_management
        self._resource_grid = resource_grid

        # Precompute indices to extract data symbols
        mask = resource_grid.pilot_pattern.mask
        num_data_symbols = resource_grid.pilot_pattern.num_data_symbols
        data_ind = tf.argsort(flatten_last_dims(mask), direction="ASCENDING")
        self._data_ind = data_ind[...,:num_data_symbols]

    def call(self, y): # pylint: disable=arguments-renamed

        # y has shape
        # [batch_size, num_rx, num_streams_per_rx, num_ofdm_symbols,...
        # ..., fft_size, data_dim]

        # If data_dim is not provided, add a dummy dimension
        if len(y.shape)==5:
            y = tf.expand_dims(y, -1)

        # Remove nulled subcarriers from y (guards, dc). New shape:
        # [batch_size, num_rx, num_rx_ant, ...
        #  ..., num_ofdm_symbols, num_effective_subcarriers, data dim]
        y = tf.gather(y, self._resource_grid.effective_subcarrier_ind, axis=-2)

        # Transpose tensor to shape
        # [num_rx, num_streams_per_rx, num_ofdm_symbols,...
        #  ..., num_effective_subcarriers, data_dim, batch_size]
        y = tf.transpose(y, [1, 2, 3, 4, 5, 0])

        # Merge num_rx amd num_streams_per_rx
        # [num_rx * num_streams_per_rx, num_ofdm_symbols,...
        #  ...,num_effective_subcarriers, data_dim, batch_size]
        y = flatten_dims(y, 2, 0)

        # Put first dimension into the right ordering
        stream_ind = self._stream_management.stream_ind
        y = tf.gather(y, stream_ind, axis=0)

        # Reshape first dimensions to [num_tx, num_streams] so that
        # we can compared to the way the streams were created.
        # [num_tx, num_streams, num_ofdm_symbols, num_effective_subcarriers,...
        #  ..., data_dim, batch_size]
        num_streams = self._stream_management.num_streams_per_tx
        num_tx = self._stream_management.num_tx
        y = split_dim(y, [num_tx, num_streams], 0)

        # Flatten resource grid dimensions
        # [num_tx, num_streams, num_ofdm_symbols*num_effective_subcarriers,...
        #  ..., data_dim, batch_size]
        y = flatten_dims(y, 2, 2)

        # Gather data symbols
        # [num_tx, num_streams, num_data_symbols, data_dim, batch_size]
        y = tf.gather(y, self._data_ind, batch_dims=2, axis=2)

        # Put batch_dim first
        # [batch_size, num_tx, num_streams, num_data_symbols]
        y = tf.transpose(y, [4, 0, 1, 2, 3])

        # Squeeze data_dim
        if y.shape[-1]==1:
            y = tf.squeeze(y, -1)

        return y
```

**Comment by kassankar** [comment]
Hi,
I made a pull request.
I hope I did it the right way.
Thanks.

**Comment by jhoydis** [comment]
Thanks. We will review it and merge it into the next release.

**Comment by gmarcusm** [comment]
Thanks, fixed in 0.9.1
